waspc/test/AppSpec/ValidTest.hs

```diff
@@ -41,6 +44,50 @@ spec_AppSpecValid = do
                          "You have more than one 'app' declaration in your Wasp app. You have 2."
                      ]
 
+    describe "'waspVersion' validation" $ do
+      describe "should validate 'waspVersion' format" $ do
+        it "returns no error if waspVersion is compatible" $ do
+          ASV.validateAppSpec basicAppSpec `shouldBe` []
+
+        it "returns an error if 'waspVersion' has an incorrect format" $ do
+          let basicAppWithInvalidWaspVersionFormat = basicApp {AS.App.wasp = AS.Wasp.Wasp {AS.Wasp.version = "0.5.2"}}
+          let basicAppDeclWithInvalidWaspVersionFormat =
+                AS.Decl.makeDecl "TestApp" basicAppWithInvalidWaspVersionFormat
+
+          ASV.validateAppSpec
+            ( basicAppSpec
+                { AS.decls = [basicAppDeclWithInvalidWaspVersionFormat]
+                }
+            )
+            `shouldBe` [ ASV.GenericValidationError
+                           "Wasp version should be in the format ^major.minor.patch"
+                       ]
+
+        it "returns an error if 'waspVersion' is not compatible" $ do
+          let incompatibleWaspVersion = DV.makeVersion (nextMajor : remaining)
+                where
+                  nextMajor = major + 1
+                  DV.Version (major : remaining) _ = Paths_waspc.version
+
+          let basicAppWithIncompatibleWaspVersion =
+                basicApp {AS.App.wasp = AS.Wasp.Wasp {AS.Wasp.version = "^" ++ DV.showVersion incompatibleWaspVersion}}
+
+          let basicAppDeclWithIncompatibleWaspVersion =
+                AS.Decl.makeDecl "TestApp" basicAppWithIncompatibleWaspVersion
+
+          ASV.validateAppSpec
+            ( basicAppSpec
+                { AS.decls = [basicAppDeclWithIncompatibleWaspVersion]
+                }
+            )
```

Comment:
This part is very similar to the same piece of code you have in the test above -> basically taking basicAppSpec and modifying setting its wasp.version field.

I would advise you either create a function for it and use it in these two places, or you modify the process of creating basicAppSpec so it takes version as an argument (but that might be a bit bigger change).